src/cmds/scripts/pbs_habitat.in

```diff
@@ -318,15 +318,31 @@ if [ "${PBS_START_SERVER:-0}" != 0 ] ; then
 			exit $ret
 		fi
 		server_started=1
+		
+		PSI_VAL=
+		# Check if PBS_SERVER_INSTANCES is set for multi-server
+		if [ ! -z "${PBS_SERVER_INSTANCES}" ]; then
+			# We want to set PBS_SERVER_INSTANCES to just the server that's being started
+			# this is needed so that qmgr doesn't try to contact other servers, which may not be up yet
+			IFS=',' read -ra PSIARR <<< "$PBS_SERVER_INSTANCES"
+			for psi_svr in "${PSIARR[@]}"; do
+				psi_hostport=(${psi_svr//:/ })
+				psi_port=${psi_hostport[1]}
+				if [[ "$psi_port" == "$PBS_BATCH_SERVICE_PORT" ]]; then
```

Comment:
how does it find the right server by just comparing port? what another server residing in a different host uses the same port...